.changeset/five-pants-jump.md

```diff
@@ -0,0 +1,17 @@
+---
+'@commercetools-docs/gatsby-theme-api-docs': major
+'@commercetools-docs/gatsby-theme-code-examples': major
+'@commercetools-docs/gatsby-theme-constants': major
+'@commercetools-docs/gatsby-theme-docs': major
+'@commercetools-website/api-docs-smoke-test': major
+---
+
+Refactored the provisioning of theme extensions' components into MDX to support components that have to be imported explicitly in MDX. Auto-provided components now have to be exported via `./shortcodes` by theme add-ons.
+
+The API documentation is not automatically injecting its components into MDX any more to allow sites with large APIs to optimize the site performance (time to interactive).
+
+To migrate, all MDX pages have to explicitly import the `ApiType` and `ApiEndpoint` components. It's recommended that sites create an intermediate `/shortcodes.js` file so that the MDX import lines only read
+
+```
+import { ApiType, ApiEndpoint } from "./shortcodes"
```

Comment:
Is there a reason for naming this `shortcodes.js`? I'm not sure what the name should imply to be honest.

If this is only meant to be used within MDX files, can we simply name it `mdx.js` or `components.js`?

Reply:
The MDX devs call  this kind of components shortcodes (e.g. https://mdxjs.com/blog/shortcodes ) - the name is derived from wordpress which many content authors know and which introduced the concept of being able to throw some sort of tag into the middle of a text document to render a more complex UI component.
https://wordpress.com/support/shortcodes/

Reply:
IMHO - if `shortcodes` is just a naming convention, I don't think we should adopt it. We generally expose modules using `index.js` across multiple projects, so sticking to that would be helpful.

Reply:
Thanks for the explanation. I didn't know about that term. As long as it's understandable and clear to the authors (assuming they are familiar with the term), it's fine by me.

Reply:
> IMHO - if `shortcodes` is just a naming convention, I don't think we should adopt it. We generally expose modules using `index.js` across multiple projects, so sticking to that would be helpful.

@davifantasia  as described in the PR description and other comments the whole point is to separate the automatically injected components from what the theme.   That's why I have to pick some naming convention and shortcodes is the one that MDX uses in their own docs.